src/device-state/target-state.ts

```diff
@@ -12,6 +11,9 @@ import { getRequestInstance } from '../lib/request';
 import * as config from '../config';
 import { writeLock } from '../lib/update-lock';
 import constants = require('../lib/constants');
+import log from '../lib/supervisor-console';
+
+class ApiResponseError extends Error {}
```

Comment:
If you're using a specific error class like this you should export it so that users can catch and check for the specific error with
```ts
try {
	...
} catch (err) {
	if (err instanceof ApiResponseError) {
		...
	}
}

Reply:
fair point